### ISSUE TITLE
Aria attributes visible for slide even after setting accessibility to false

### ISSUE THREAD

**Issue by sunnydavera** [kenwheeler/slick#1820]
Hi,
This is an amazing carousel and we are actively using it. However, I have discovered that the aria-hidden attributes are set for the slides even if the accessibility flag is set to false. I think this is incorrect.
We have a requirement from ADA perspective to allow only tabbing through each slide without tabbing on the arrow buttons. I can set the tabindex for these buttons programmatically to -1 and prevent the tabbing but since the aria-hidden is defaulted to true when carousel is initialized, the invisible slides are never read even though the user can tab through them and are visible now.
Why not set the aria tags only on basis of the accessibility flag to allow the flexibility of setting them based on user requirement?

**Comment by qbunt** [comment]
Same issue as https://github.com/kenwheeler/slick/issues/1544. May be faster to fix and PR on your own.

**Comment by ahmadalfy** [comment]
The `accessibility` option only control tabbing and arrow key navigation. Perhapse we need another option to enable/disable aria attributes?

**Comment by qbunt** [comment]
As @sunnydav mentions above, adding ARIA attributes of any kind are going to influence ARIA based screen readers across the board. It seems redundant to have a second setting that would turn these attributes on/off when really ARIA attributes should only be added/removed when this option is set to 'true'.

This is a real problem when you need to override the ARIA attributes because Slick doesn't implement what you need, there's no real way to de-activate them completely. Not all accessibility requirements are asking for the same thing.

This would be my vote:
`accessibility:false;` -> no ARIA tags on anything
`accessibility:true;` -> ARIA tags where needed, using 'aria-hidden' very carefully

**Comment by qbunt** [comment]
It may actually be smarter to remove this implementation completely rather than leave it the way it is.

**Comment by ahmadalfy** [comment]
@qbunt can you please explain what is wrong with the current implementation and I will see what we can do to fix it?

**Comment by qbunt** [comment]
The `aria-hidden=true` attribute on elements (even when set to `accessibility:false` will be evaluated with screen reader software in a way that's not desirable. Because that property appears with the `accessibility` config true or false, if *you need to change* this, you have to loop through every slide and remove the `aria-hidden` attribute manually on every change. It makes supporting WCAG standards (the ADA compliance @sunnydav mentions) really challenging.

If you set `accessibility:false` there should be *no ARIA tags on any element* by default. The `aria-hidden`[property is explained here pretty well](https://www.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/).

If you are attempting to use Slick in an environment where you have to support WCAG standards (banking, e-commerce, retail, etc), Slick makes this nearly impossible with the current implementation, as the ARIA attributes are not compliant. [WCAG 2.0 standards for carousels are well established.](http://www.w3.org/WAI/tutorials/carousels/functionality/)

My point is, either remove all the ARIA attributes if `accessibility:false` *or* remove the entire implementation, as it makes supporting the published standards far more work than it should be.

**Comment by cshold** [comment]
I agree, the aria implementation could use some work. I've opened a new issue about `aria-live="polite"` [here](https://github.com/kenwheeler/slick/issues/1937) for those interested.

**Comment by ahmadalfy** [comment]
@cshold :+1:

**Comment by qbunt** [comment]
Hey @ahmadalfy, any update on this, I know this is likely not high on the priority list, but it'd be great to hear if it's going to get some love or if a PR is needed to close this out. Cheers.

**Comment by JoshuaKGoldberg** [comment]
This is still an issue, and is still relevant. An opt-out mode for `aria-hidden` is, ironically, really useful for making usage of this library accessible.

**Comment by karlchelton** [comment]
An opt out feature needs to be added.. for the sake of accessibility and audit reviews.

**Comment by fisher2470** [comment]
Also currently experiencing this issue, an opt-out is needed

**Comment by JohnMDDD** [comment]
I also am in need of a solution to this problem with ADA compliance. An opt-out is needed!
